### PR TITLE
Add get_ref to EntityRef

### PR DESCRIPTION
**Pull request by nicopap**
# Objective

To mirror the `Ref` added as `WorldQuery`, and the `Mut` in `EntityMut::get_mut`, we add `EntityRef::get_ref`, which retrieves `T` with tick information, but *immutably*.

## Solution

- Add the method in question, also add it to`UnsafeEntityCell` since this seems to  be the best way of getting that information.

Also update/add safety comments to neighboring code.

---

## Changelog

- Add `EntityRef::get_ref` to get an `Option<Ref<T>>` from `EntityRef`

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Useful. Style is consistent, docs are good, safety comments make sense.

**Review by JoJoJet — APPROVED** [review]

File: crates/bevy_ecs/src/world/entity_ref.rs [hunk 1]
```diff
@@ -121,6 +121,16 @@ impl<'w> EntityRef<'w> {
         unsafe { self.as_unsafe_world_cell_readonly().get::<T>() }
     }
 
+    /// Gets access to the component of type `T` for the current entity,
+    /// including change detection information in [`Ref`].
```
Comment (james7132) [inline crates/bevy_ecs/src/world/entity_ref.rs:125]:
```suggestion
    /// including change detection information as a [`Ref`].
```

File: crates/bevy_ecs/src/world/unsafe_world_cell.rs [hunk 2]
```diff
@@ -651,15 +651,14 @@ impl<'w> UnsafeEntityCell<'w> {
...
-        // - entity location is valid
-        // - proper world access is promised by caller
+        // - `storage_type` is correct (T component_id + T::STORAGE_TYPE)
+        // - `location` is valid
+        // - proper aliasing is promised by caller
         unsafe {
             get_component(
                 self.world,
-                component_id,
+                self.world.components().get_id(TypeId::of::<T>())?,
```
Comment (james7132) [inline crates/bevy_ecs/src/world/unsafe_world_cell.rs:661]: Why was this moved back inside the unsafe block? It's still safe.

Comment (nicopap) [reply crates/bevy_ecs/src/world/unsafe_world_cell.rs:661]: I thought it was more explicit. It's a bit more obvious that the code is safe, one of the invariants is that "`component_id` should be that of `T`". With this change, we can see `TypeId::of::<T>` _inline_ where it's relevant, rather than somewhere else.

Though of course it's a minor improvement, since the somewhere else in question is not that far off. It wasn't mindless. But I can revert it.

**Review by james7132 — APPROVED** [review]

**Review by mockersf — APPROVED** [review]
